src/lib.rs

```diff
@@ -19,7 +19,14 @@
 //! base64-encoded DER, PEM objects are delimited by header and footer lines which indicate the type
 //! of object contained in the PEM blob.
 //!
-//! The [rustls-pemfile](https://docs.rs/rustls-pemfile) crate can be used to parse PEM files.
+//! Types here can be created from:
+//!
+//! - DER using (for example) [`PrivatePkcs8KeyDer::from`].
+//! - PEM using (for example) [`PrivatePkcs8KeyDer::decode_from_pem`] via the [`DecodePem`] extension trait.
+//!
+//! `decode_from_pem` returns the first matching item from the given input.
+//! It is usual for given PEM file to contain multiple items: if you wish
+//! to examine all of these you can use the iterator-based API in the [`pem`] module.
```

Comment:
I think we should have a convenience API that fits the use case of easily getting a `Vec<CertificateDer<'static>>` from a file, and this shouldn't require going via the `Item` API.

Reply:
Does `Vec<CertificateDer<'static>>::decode_from_pem_reader(BufReader::new(File::open("certs.pem")?))` (modulo name of that function) meet that?

Maybe we could add a decode-from-io::Read, reducing that to `Vec<CertificateDer<'static>>::decode_from_pem_reader(&mut File::open("certs.pem")?)` and hiding the BufReader requirement ?

Reply:
> Does `Vec<CertificateDer<'static>>::decode_from_pem_reader(BufReader::new(File::open("certs.pem")?))` (modulo name of that function) meet that?

I feel like writing anything like `Vec<FooBar>::method()` is pretty unusual to write/read, though I guess `Vec::from_pem_reader()` would work as long as there is type inference? Might be fine.

> Maybe we could add a decode-from-io::Read, reducing that to `Vec<CertificateDer<'static>>::decode_from_pem_reader(&mut File::open("certs.pem")?)` and hiding the BufReader requirement ?

Paving the cow path for `File` would be nice, I think.